### PR TITLE
Support pro plans

### PR DESCRIPTION
**Pull request by macav**
Currently the API calls always end up on `https://postgres-starter-api.heroku.com`. This means that if DB used has a professional plan, the requests return 404. 

Inspired by the official CLI: https://github.com/heroku/cli/blob/master/packages/pg-v5/commands/backups/capture.js

I had to include the `platform-api` gem and perform a call to fetch addons of an app, so I can determine the plan of the DB addon.

### CODE REVIEW

File: lib/heroku/api/postgres/client.rb [hunk 1]
```diff
@@ -19,46 +21,38 @@ def databases
...
         end
 
-        def host_for(database)
-          starter_plan?(database) ? 'https://postgres-starter-api.heroku.com' : 'https://postgres-api.heroku.com'
-        end
-
-        def starter_plan?(database)
-          database[:plan].match(/(dev|basic)$/)
-        end
-
```
Comment (macav) [inline lib/heroku/api/postgres/client.rb:46]: moved under `Databases`

File: lib/heroku/api/postgres/client.rb [hunk 2]
```diff
@@ -19,46 +21,38 @@ def databases
...
+        def add_auth_headers(req)
           req['Accept'] = 'application/vnd.heroku+json; version=3'
           req.basic_auth '', @oauth_client_key
         end
 
         def start_request(req, url)
           http_new = Net::HTTP.new(url.hostname, url.port)
           http_new.use_ssl = true
-          response = http_new.start { |http| http.request(req) }
+          http_new.start { |http| http.request(req) }
```
Comment (macav) [inline lib/heroku/api/postgres/client.rb:55]: rubocop fixes

File: spec/heroku/api/postgres/backups_spec.rb [hunk 3]
```diff
@@ -19,8 +19,9 @@
   end
 
   describe '#schedules' do
+    let(:app_id) { ENV['VALID_APP_ID'] }
     let(:database_id) { ENV['VALID_DATABASE_ID_WITH_SCHEDULES'] }
-    subject(:json_response) { client.backups.schedules(database_id) }
+    subject(:json_response) { client.backups.schedules(app_id, database_id) }
```
Comment (macav) [inline spec/heroku/api/postgres/backups_spec.rb:24]: i adapted the cassettes

**Review by coorasse — APPROVED** [review]
Thanks!

**Pull request merged**